src/lib/ip-whitelist.js

```diff
@@ -0,0 +1,42 @@
+const logger = require('@financial-times/n-logger').default;
+const fetchres = require('fetchres');
+const ip = require('ip');
+
+const backupWhitelist = require('./ip-whitelist-backup.json').addresses;
+
+const whitelistUrl = 'https://api.fastly.com/public-ip-list';
+
+function IpWhitelist () {
+	this.fetchedWhitelist = null;
+	this.poll();
+	setInterval(() => this.poll(), 10000); // every 10 seconds
```

Comment:
aye, I didn't particularly like the look of `ft-poller` - it doesn't have the main logic in one place and also it doesn't do any logging... it's smelly :)